flepimop/gempyor_pkg/src/gempyor/distributions.py

```diff
@@ -72,6 +73,18 @@ def _sample_from_generator(
         """
         raise NotImplementedError
 
+    @property
+    @abstractmethod
+    def support(self) -> tuple[float | int, float | int]:
+        """
+        The theoretical support of the distribution as a (min, max) tuple.
+
+        Returns:
+            A tuple representing the upper and lower bounds in the distribution's
+            support.
+        """
+        raise NotImplementedError
+
```

Comment:
I think we'd be better served here by adopting the wrapping delegate implementers - so have this return `(..., ...)` and then the subclasses indicate their lower / upper bounds.

is this something w/ can do with `ClassVar` definitions here, then values supplied in the sub classes - will pydantic enforce that developers define them in subclasses? ala, https://docs.pydantic.dev/latest/concepts/models/#class-variables